railties/lib/rails/generators/rails/scaffold/scaffold_generator.rb

```diff
@@ -12,16 +12,20 @@ class ScaffoldGenerator < ResourceGenerator # :nodoc:
       class_option :assets, type: :boolean
       class_option :resource_route, type: :boolean
       class_option :scaffold_stylesheet, type: :boolean
+      class_option :skip_system_test, type: :boolean, default: false,
+                                      desc: "Skip system test files"
 
       def handle_skip
         @options = @options.merge(stylesheets: false) unless options[:assets]
         @options = @options.merge(stylesheet_engine: false) unless options[:stylesheets] && options[:scaffold_stylesheet]
-        @options = @options.merge(system_tests: false) if options[:api]
+        @options = @options.merge(skip_system_test: true) if options[:api]
       end
 
       hook_for :scaffold_controller, required: true
 
-      hook_for :system_tests, as: :system
+      hook_for :system_tests, as: :system do |system_test|
+        invoke system_test unless options[:skip_system_test]
+      end
```

Comment:
I think the change here is what you meant when you said "thor can skip without the extra configuration".

That's true and this is working on master. The problem comes in in #29418 because it no longer uses the `hook_for` since that would load the default plain system test template and we want to load the scaffold system test template with create/update/delete actions just like the controller test file does. In that case we need to pass the `class_option` to the scaffold generator. This was where I noticed that the config file was using `system_tests` as the `option` and not `skip_system_test`.

So I can either leave the config as is and pass `system_test` as a class_option to the scaffold generator, or change it as I did here.